src/api/libcellml/types.h

```diff
@@ -70,4 +74,133 @@ using UnitsPtr = std::shared_ptr<Units>; /**< Type definition for shared units p
 class Variable; /**< Forward declaration of Variable class. */
 using VariablePtr = std::shared_ptr<Variable>; /**< Type definition for shared variable pointer. */
 
+class Unit;
+using UnitPtr = std::shared_ptr<Unit>;
+
+/**
+ * @brief The Unit class
+ *
+ * The Unit class can be used to refer to a unit in a @ref Units.
+ */
+class LIBCELLML_EXPORT Unit
+{
+public:
+    ~Unit(); /**< Destructor. */
+    Unit() = delete; /**< Constructor. */
+    Unit(const Unit &rhs) = delete; /**< Copy constructor. */
+    Unit(Unit &&rhs) noexcept = delete; /**< Move constructor. */
+    Unit &operator=(Unit rhs) = delete; /**< Assignment operator. */
+
+    /**
+     * @brief Create a unit reference object.
+     *
+     * Factory method to create a @ref UnitPtr.  Create a unit with @ref Units
+     * and index with::
+     *
+     *   auto unit = libcellml::UnitPtr::create(units, index);
+     *
+     * @return A smart pointer to a @ref UnitPtr object.
+     */
+    static UnitPtr create(const UnitsPtr &units, size_t index) noexcept;
+
+    /**
+     * @brief Get the unit.
+     *
+     * Get the unit in the @ref Units.
+     *
+     * @return The unit.
+     */
+    UnitsPtr units() const;
```

Comment:
Why is this function called `units` when it returns a `unit` ?

Reply:
Or ... what does this function really do?  Return the pointer to the `Units` which this `Unit` depends on? Either way, the docs don't match what the function does.

Reply:
It returns a `Units`, so if anything the documentation should be fixed to read: `@return The units.`?